drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -0,0 +1,140 @@
+
+#include <stdexcept>
+#include "RigidBodySystem.h"
+#include "RigidBodyIK.h"  // required for resolving initial conditions
+#include "urdfParsingUtil.h"
+
+using namespace std;
+using namespace Eigen;
+using namespace Drake;
+
+
+RigidBodySystem::StateVector<double> Drake::getInitialState(const RigidBodySystem& sys) {
+
+  VectorXd x0 = Matrix<double,Dynamic,1>::Random(sys.tree->num_positions+sys.tree->num_velocities);
+
+  if (sys.tree->getNumPositionConstraints()) {
+    // todo: move this up to the system level?
+
+    std::vector<RigidBodyLoop,Eigen::aligned_allocator<RigidBodyLoop> > const& loops = sys.tree->loops;
+
+    int nq = sys.tree->num_positions;
+    int num_constraints = 2*loops.size();
+    RigidBodyConstraint** constraint_array = new RigidBodyConstraint*[num_constraints];
```

Comment:
that's what inverseKin asks for.  http://drake002.csail.mit.edu/drake/doxygen_cpp/inverse_kin_8cpp.html#a9ea7c55639588ecd09ec9b928e304e11 
and since the lifetime is trivial it seems ok.  The only downside I can see is if an exception was thrown in inverseKin and caught above me, then I would leak memory.
i think we should definitely update the interface to inverseKin eventually.